File: max_ascending_1800.cpp
#include <iostream>
#include <vector>
using namespace std;

int maxAscendingSum(vector<int> nums) {
    int l = 0;
    int sum = nums[l];
    int max_sum=sum;
    for (int r=1;r<nums.size();r++){
        if(nums[r]>nums[r-1]){
            sum+=nums[r];
        }
        else{
            l=r;
            sum=nums[l];
            max_sum=max(sum, max_sum);
        }
    }
    return max_sum;
}
int main(){
    vector<int> v={12,17,15,13,10,11,12};
    cout<<maxAscendingSum(v);
}

File: max_ascending_1800.py
def maxAscendingSum(nums):
    l = 0
    sum = nums[l]
    max_sum=sum
    for r in range(1, len(nums)):
        if nums[r] > nums[r - 1]:
            sum += nums[r]
        else:
            l = r
            max_sum=max(sum,max_sum)
            sum = nums[l]
            max_sum=max(sum,max_sum)

    return max(sum,max_sum)
print(maxAscendingSum([12,17,15,13,10,11,12]))